packages/unchained-client/src/utxo/parser/thorchain.ts

```diff
@@ -0,0 +1,249 @@
+import type { Axios } from 'axios'
+import axios from 'axios'
+
+import { type BaseTxMetadata, Dex, TradeType } from '../../types'
+import type { SubParser, Tx, TxSpecific } from '../parser'
+
+export type LiquidityType = 'Savers' | 'LP'
+export type SwapType = 'Standard' | 'Streaming'
+
+interface Liquidity {
+  type: LiquidityType
+}
+
+interface Swap {
+  type: SwapType
+}
+
+export interface TxMetadata extends BaseTxMetadata {
+  parser: 'thorchain'
+  memo: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+interface Coin {
+  amount: string
+  asset: string
+}
+
+interface InOut {
+  txID: string
+  address: string
+  coins: Coin[]
+  height?: string
+}
+
+interface Metadata {
+  addLiquidity?: {
+    liquidityUnits: string
+  }
+  refund?: {
+    affiliateAddress: string
+    affiliateFee: string
+    memo: string
+    networkFees: Coin[]
+    reason: string
+  }
+  swap?: {
+    affiliateAddress: string
+    affiliateFee: string
+    isStreamingSwap: boolean
+    liquidityFee: string
+    networkFees: Coin[]
+    swapSlip: string
+    swapTarget: string
+    memo: string
+  }
+  withdraw?: {
+    asymmetry: string
+    basisPoints: string
+    impermanentLossProtection: string
+    liquidityUnits: string
+    memo: string
+    networkFees: Coin[]
+  }
+}
+
+interface Actions {
+  date: string
+  height: string
+  in: InOut[]
+  out: InOut[]
+  metadata?: Metadata
+  pools: string[]
+  status: 'success' | 'pending'
+  type: 'swap' | 'addLiquidity' | 'withdraw' | 'donate' | 'refund' | 'switch'
+}
+
+export interface ActionsResponse {
+  actions: Actions[]
+  count: string
+  meta: {
+    nextPageToken: string
+    prevPageToken: string
+  }
+}
+
+interface ExtraMetadata {
+  method: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+const getLiquidityType = (pool: string): LiquidityType => (pool.includes('/') ? 'Savers' : 'LP')
+
+const getSwapType = (memo: string): SwapType => {
+  const regex = /:\d+\/\d+\/\d+:/
```

Comment:
We may want to make it clear here that this regex tests streaming swap memos, perhaps extract it as a const outside of this scope so it can be more easily visually/grepped for

Reply:
https://github.com/shapeshift/web/pull/5986/commits/efc20f05273df748fdf20442cea2874f23639a30